creme/stream/datasets/agrawal.py

```diff
@@ -0,0 +1,316 @@
+import numpy as np
+
+from . import base
+from creme.utils.skmultiflow_utils import check_random_state
+
+
+class AGRAWALGenerator(base.SyntheticDataset):
+    """Agrawal stream generator.
+
+    The generator was introduced by Agrawal et al., and was common source
+    of data for early work on scaling up decision tree learners.
+    The generator produces a stream containing nine features, six numeric and
+    three categorical.
+    There are ten functions defined for generating binary class labels from the
+    features. Presumably these determine whether the loan should be approved.
+    The features and functions are listed in the original paper [1]_.
+
+    | feature name | feature description  | values                                                              |
+    |--------------|----------------------|---------------------------------------------------------------------|
+    | salary       | the salary           | uniformly distributed from 20k to 150k                              |
+    | commission   | the commission       | if (salary < 75k) then 0 else uniformly distributed from 10k to 75k |
+    | age          | the age              | uniformly distributed from 20 to 80                                 |
+    | elevel       | the education level  | uniformly chosen from 0 to 4                                        |
+    | car          | car maker            | uniformly chosen from 1 to 20                                       |
+    | zipcode      | zip code of the town | uniformly chosen from 0 to 8                                        |
+    | hvalue       | value of the house   | uniformly distributed from 50k x zipcode to 100k x zipcode          |
+    | hyears       | years house owned    | uniformly distributed from 1 to 30                                  |
+    | loan         | total loan amount    | uniformly distributed from 0 to 500k                                |
+
+    Parameters
+    ----------
+    classification_function: int (Default=0)
```

Comment:
You don't have to specify types and default values. They are extracted from the `__init__` signature.

Reply:
Old habits die hard :-) 
I removed types and default values.